### DISCUSSION TITLE
Is it possible to have a feedback (reinforcement) loop

### DISCUSSION THREAD

**Discussion by afonari** [deepmodeling/deepmd-kit#28] [Q&A] [answered]
The idea is to run MD using model. In the case if uncertainty of the model is high (as evaluated from Gaussian kernel or similar) call backend (using iPI for example) and to perform the actual calculation, and use the results to update the model.

**Comment by jameswind** [comment, marked as answer]
We are on the way of releasing the code for this paper:
"Active learning of uniformly accurate interatomic potentials for materials
simulation." *Physical Review Materials* 3.2 (2019): 023804.
which will meet your purpose.
This would take one or two more months since we are stilling testing it and
preparing better documentation.

For now, I think you could also realize it in a more manual way:
1) during training you could obtain a few, say 4, different models by
changing the seed in the training input file;
2) we have an interface with the LAMMPS code, which prints out the
uncertainty, we call model deviation, on the fly.
After training, one may print out the model deviation when running MD with
LAMMPS. For example, if one has trained 2 models, `graph.00.pb` and
`graph.01.pb`, one may use the following script
```bash
pair_style  deepmd graph.00.pb graph.01.pb 10 model_devi.out
pair_coeff
```
Here `graph.00.pb` and `graph.01.pb` are served as a list. Doing so LAMMPS
will print the model deviation in terms of energies and forces in the file
`model_devi.out`, in every `10` MD steps.
3) definitely, you could do some sampling using i-PI or any other MD
engines and test the model deviation when you obtain a bunch of
trajectories.

**Comment by afonari** [comment]
Thanks a fast reply. I will study the paper.
